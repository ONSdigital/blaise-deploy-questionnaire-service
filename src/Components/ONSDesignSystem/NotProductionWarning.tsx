import React, {ReactElement} from "react";


function NotProductionWarning(): ReactElement {
    return (
        <>
            <div className="census-warning" style={{background: "#d0021b", color: "#d0021b"}}>
                <div className="container">
                    <div className="panel panel--warn-branded panel--no-title" style={{background: "#d0021b"}}>
                        <span className="panel__icon" aria-hidden="true" style={{color: "#d0021b"}}>!</span>
                        <div className="panel__body">
                            <p className="u-mb-no">This environment is not a production environment. Do not upload any live data to this service.</p>
                        </div>
                    </div>
                </div>
            </div>
        </>
    );
}

export default NotProductionWarning;
