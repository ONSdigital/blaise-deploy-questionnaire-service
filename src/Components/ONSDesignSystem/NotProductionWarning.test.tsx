import React from "react";
import Enzyme, {shallow, render} from "enzyme";

import Adapter from "enzyme-adapter-react-16";
import NotProductionWarning from "./NotProductionWarning";

describe("ONS Not production warning Test", () => {
    Enzyme.configure({ adapter: new Adapter() });

    it("matches Snapshot", () => {
        expect(render(<NotProductionWarning/>)).toMatchSnapshot();
    });

    it("should render correctly", () => expect(shallow(<NotProductionWarning/>).exists()).toEqual(true));
});
