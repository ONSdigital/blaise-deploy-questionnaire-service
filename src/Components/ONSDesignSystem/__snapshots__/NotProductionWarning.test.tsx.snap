// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ONS Not production warning Test matches Snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "census-warning",
      "style": "background:#d0021b;color:#d0021b",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "container",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "panel panel--warn-branded panel--no-title",
              "style": "background:#d0021b",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "aria-hidden": "true",
                  "class": "panel__icon",
                  "style": "color:#d0021b",
                },
                "children": Array [
                  Object {
                    "data": "!",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "span",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "panel__body",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "class": "u-mb-no",
                      },
                      "children": Array [
                        Object {
                          "data": "This environment is not a production environment. Do not upload any live data to this service.",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "p",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                      },
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "aria-hidden": undefined,
                  "class": undefined,
                  "style": undefined,
                },
                "x-attribsPrefix": Object {
                  "aria-hidden": undefined,
                  "class": undefined,
                  "style": undefined,
                },
              },
              Object {
                "attribs": Object {
                  "class": "panel__body",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "class": "u-mb-no",
                    },
                    "children": Array [
                      Object {
                        "data": "This environment is not a production environment. Do not upload any live data to this service.",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "p",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "aria-hidden": "true",
                    "class": "panel__icon",
                    "style": "color:#d0021b",
                  },
                  "children": Array [
                    Object {
                      "data": "!",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "span",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "aria-hidden": undefined,
                    "class": undefined,
                    "style": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "aria-hidden": undefined,
                    "class": undefined,
                    "style": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
              "style": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
              "style": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
